champ/lib/lulw/lib.rs

```diff
@@ -31,13 +33,20 @@ pub enum WalletError {
     DecryptionError(String),
 }
 
-pub fn generate_wallet(password: String) -> Result<String, WalletError> {
+pub type WalletAndAddress = (String, String);
+
+pub fn generate_wallet(password: String) -> Result<WalletAndAddress, WalletError> {
     debug!("generating wallet");
-    let (ciphertext, salt, nonce) = {
+    let (public_key, (ciphertext, salt, nonce)) = {
         let private_key = crypto::signatures::ed25519::generate_private_key()
             .map_err(|e| WalletError::GeneratePrivateKeyError(e.to_string()))?;
 
-        encrypt(&private_key, password.as_bytes()).map_err(|e| WalletError::EncryptionError(e.to_string()))?
+        let public_key = crypto::signatures::ed25519::create_public_key(&private_key)
+            .map_err(|e| WalletError::CreatePublicKeyError(e.to_string()))?;
+        (
+            public_key,
+            encrypt(&private_key, password.as_bytes()).map_err(|e| WalletError::EncryptionError(e.to_string()))?,
+        )
```

Comment:
```suggestion
        let encrypted_wallet = encrypt(&private_key, password.as_bytes()).map_err(|e| WalletError::EncryptionError(e.to_string()))?;
        let public_key = crypto::signatures::ed25519::create_public_key(&private_key)
            .map_err(|e| WalletError::CreatePublicKeyError(e.to_string()))?;
        (public_key, encrypted_wallet)
```
I think this might look a bit overwhelming otherwise with the nested tupel (might not be perfectly following rustfmt)